<!-- tutorial.html -->
{% load static %}

<div class="row toggle toggle-default" id="tutorial">

  <div class="pagetitle mb-4">
    <h1>Tutorial</h1>
  </div>

  <div class="col-sm-6">
    <div class="card" style="height: 15rem;">
      <div class="card-body">
        <h5 class="card-title fs-4">About sRNA Analyst</h5>
        <ul>
          <li class="my-1"><a href="#what-is-sRNA-Analyst?" class="fs-5">What is sRNA Analyst?</a></li>
          <li class="my-1"><a href="#motivation-of-developing-sRNA-Analyst" class="fs-5">Motivation of developing sRNA
              Analyst</a></li>
          <li class="my-1"><a href="#impact-of-sRNA-Analyst" class="fs-5">Impact of sRNA Analyst</a></li>
        </ul>
      </div>
    </div>
  </div>

  <div class="col-sm-6">
    <div class="card" style="height: 15rem;">
      <div class="card-body">
        <h5 class="card-title fs-4">Using sRNA Analyst</h5>
        <ul>
          <li class="my-1"><a href="#Quick-Start" class="fs-5">Quick Start</a></li>
          <li class="my-1"><a href="#Preprocess-Page" class="fs-5">Preprocess Page</a></li>
          <li class="my-1"><a href="#Analysis-Page" class="fs-5">Analysis Page</a></li>
          <li class="my-1"><a href="#Database-Page" class="fs-5">Database Page</a></li>
        </ul>
      </div>
    </div>
  </div>

  <div class="col-12" id="what-is-sRNA-Analyst?">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">What is sRNA Analyst?</h5>
        <p>sRNA Analyst is an pipeline analysis tool optimized for sRNA-seq data, including a complete
          data preprocessing workflow, downstream analysis tools, and a collection of sRNA references and
          experimental datasets.</p>
        <div class="row">
          <div class="col-md-5">
            <img src="/sRNAanalyst_static/img/Browser.png" class="img-fluid rounded-start pt-1">
          </div>
          <div class="col-md-7">
            <ul>
              <li class="my-1">The preprocessing tools currently offer a basic workflow for small RNA, including adapter
                trimming, quality control, length filtering, normalization, and reference alignment. It provides
                additional helper tools for processing reference files or converting a SAM/BAM file to CSV format for
                downstream analysis.</li>
              <li class="my-1">The downstream analysis tools include Transcriptome Browser, Metagene, Region Density
                Distribution, Position (head/tail/start codon/stop codon) Readcount Distribution, Fold-change, and
                Scatter Plot, assisting in observing the distribution of small RNA on specific target RNA under
                different
                datasets.</li>
              <li class="my-1">The database contains experimental datasets primarily focused on C.elegans, along with
                common model organism miRNA, mRNA references, and partial target gene lists of worm, including Soma
                target, Germline target, WAGO and CSR-1 target.</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>

  <div class="col-12" id="motivation-of-developing-sRNA-Analyst">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">Motivation of developing sRNA Analyst</h5>
        <p>Up to the present, there have been numerous pipeline analysis tools for small RNA, with a predominant focus
          on analyzing miRNA and piRNA. However, there is a lack of web-based tools capable of rapidly generating
          analyses such as metagene profiles, region density, or position distributions. Moreover, there is a scarcity
          of tools that can independently handle both data preprocessing and downstream analysis. Therefore, we
          developed a tool to assist biologists to swiftly perform data preprocessing, RNA profiling, identify
          differentially expressed transcripts, or independently conduct other functional analyses directly from
          experimental FASTQ or SAM/BAM files.</p>
      </div>
    </div>
  </div>

  <div class="col-12" id="impact-of-sRNA-Analyst">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">Impact of sRNA Analyst</h5>
        <p>sRNA Analyst is the first web-based analysis tool capable of examining small RNA metagene profiles, region
          density, and position distributions. It provides a user-friendly interface that allows users to click and
          perform data preprocessing for sRNA-seq or use other pipelines that can output SAM/BAM files. With just the
          provision of an intermediate CSV file, users can conduct downstream functional analyses anytime, anywhere.
          This enables the observation of sRNA abundance, differential expression, or the examination of distribution
          and fold-change variations at the transcript level for specific target lists across different datasets.</p>
      </div>
      <img src="/sRNAanalyst_static/img/workflow.png" class="mx-auto w-75">
      <div class="row px-5 py-3">
        <div class="col-md-3"><img src="/sRNAanalyst_static/img/Metagene.png" class="img-fluid"></div>
        <div class="col-md-3"><img src="/sRNAanalyst_static/img/Stop_Codon.png" class="img-fluid"></div>
        <div class="col-md-3"><img src="/sRNAanalyst_static/img/Fold_Change.png" class="img-fluid"></div>
        <div class="col-md-3"><img src="/sRNAanalyst_static/img/Scatter.png" class="img-fluid"></div>
      </div>
    </div>
  </div>

  <div class="col-12" id="Quick-Start">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">Quick Start</h5>
        <ul>
          <li class="my-1">Navigate to the "Preprocess Page," choose the workflow in the "Toolbox," and upload your
            compressed FASTQ file in the settings. Then, individually select your desired parameters for Trimming,
            Normalization, and Mapping. Once configured, click "Submit" and wait a few minutes until the results appear.
          </li>
          <img src="/sRNAanalyst_static/img/demo1.png" class="mx-auto w-75 img-thumbnail mb-3">
          <li class="my-1">Download the final CSV file. After entering the "Analysis Page," select "Input > Dataset" and
            upload the CSV file based on your requirements for single or paired data. Choose "Input > Target List" and
            upload the target transcript list for analysis, or use the default "All Target RNA."</li>
          <img src="/sRNAanalyst_static/img/demo2.png" class="mx-auto w-50 img-thumbnail mb-2">
          <img src="/sRNAanalyst_static/img/demo3.png" class="mx-auto w-75 img-thumbnail mb-3">
          <li class="my-1">Click on various tools in the "Toolbox" on the "Analysis Page" and configure them as needed.
            You may need to upload target RNA references or other files. Once configured, click "Submit" and wait a few
            minutes until the results appear.</li>
          <img src="/sRNAanalyst_static/img/demo4.png" class="mx-auto w-75 img-thumbnail mb-3">
          <p><b>Note:</b> You can load examples to observe results, or download data from the database for personal use.
            To review the analysis results, you can input the User ID from that session to load all settings and
            outcomes. All records will be stored on the server for 3 days unless updated.</p>
          <img src="/sRNAanalyst_static/img/demo5.png" class="mx-auto w-50 img-thumbnail mb-3">
          <img src="/sRNAanalyst_static/img/demo6.png" class="mx-auto w-75 img-thumbnail mb-3">
        </ul>
      </div>
    </div>
  </div>

  <div class="col-12" id="Preprocess-Page">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">Preprocess Page</h5>
        <p>
          The Preprocess page includes two tools: <b>Workflow</b> and <b>Helper Tool</b>. Workflow facilitates common
          preprocessing steps for NGS data, including adapter trimming, quality control, length filtering,
          normalization, and reference alignment. The Helper Tool provides eight data processing functions for
          individual files, commonly used for tasks such as converting SAM/BAM to CSV files or handling reference
          files, such as converting ncRNA to miRNA.
        </p>
        <p>The Workflow output includes:</p>
        <ul>
          <li class="my-1">
            Reports before and after FastQC, along with quality box-plots for each position, aiding in assessing whether
            trimming results meet expectations.
          </li>
          <li class="my-1">
            Normalization Factor
          </li>
          <li class="my-1">
            De-duplicated file <b>[data]_collapse.fa</b>
          </li>
          <li class="my-1">
            Bowtie2 alignment results, including alignment ratios, site and read-count quantities, and
            <b>[data]_mapped.sam</b>
          </li>
          <li class="my-1">
            Organized alignment results <b>[data].csv</b>, suitable as dataset for downstream analysis
          </li>
        </ul>
        <p>
          The Helper Tool includes functions such as file conversion, ID tag detection, nucleotide conversion, reverse
          complement, head/tail or length filtering, collapsing, read-count distribution or normalization, and column
          merging from different files. Processed files will have the suffix "_preprocess" appended to the filename.
        </p>
        <p>For detailed usage instructions, hover over the tool options or refer to examples.</p>
      </div>
    </div>
  </div>

  <div class="col-12" id="Analysis-Page">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">Analysis Page</h5>
        <p>
          The Analysis page includes seven analytical functionalities, requiring users to specify the <b>Dataset</b> and
          the
          <b>Target List</b> of interest to activate them.
        </p>
        <ul>
          <li class="my-1">
            <b>Browser:</b> Visualizes read-reference alignment results, with users providing a target RNA reference for
            comparison. Users can further define specific regions of interest (e.g., UTR, CDS, exon, intron). Analysis
            results include table view and graphical view, allowing for detailed examination of small RNA binding sites,
            read-count distribution, and alignment details on specific target RNAs.
            <br>
            <img src="/sRNAanalyst_static/img/Browser.png" class="mx-auto w-50 my-3">
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Density:</b> Compares read-count density among different datasets, target lists, or regions (e.g., 5'UTR,
            CDS,
            3'UTR) using box plots. This is useful for observing binding trends of small RNAs across different targets,
            supplemented with statistical tests (including T-test and U-test).
            <br>
            <img src="/sRNAanalyst_static/example/job/analysis/Density_0.png" class="mx-auto w-75 my-3">
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Metagene:</b> Presents overall binding trends in different datasets and target lists via line graphs.
            Dividing the
            target RNA into 100%, this visualization observes the distribution of sRNA read-counts and binding ratios.
            For
            example, it can assess whether CSR-1 targets preferentially bind near the 3' end compared to WAGO targets in
            PRG-1 mutant.
            <br>
            <div class="row my-3">
              <div class="col-md-5">
                <img src="/sRNAanalyst_static/example/job/analysis/Metagene_1.png" class="img-fluid rounded-start">
              </div>
            </div>
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Head & Tail:</b> Displays read-count distribution near the head and tail within a specific distance via
            line
            graphs, allowing for comparisons among different datasets or target lists.
            <br>
            <div class="row my-3">
              <div class="col-md-5">
                <img src="/sRNAanalyst_static/img/Head.png" class="img-fluid rounded-start">
              </div>
              <div class="col-md-5">
                <img src="/sRNAanalyst_static/img/Tail.png" class="img-fluid rounded-start">
              </div>
            </div>
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Start & Stop Codon:</b> Shows read-count distribution near the start and stop codons within a specific
            distance
            via line graphs, facilitating comparisons among different datasets or target lists.
            <br>
            <div class="row my-3">
              <div class="col-md-5">
                <img src="/sRNAanalyst_static/img/Start_Codon.png" class="img-fluid rounded-start">
              </div>
              <div class="col-md-5">
                <img src="/sRNAanalyst_static/img/Stop_Codon.png" class="img-fluid rounded-start">
              </div>
            </div>
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Fold Change:</b> Compares fold-changes in sRNA read-counts binding to target RNAs between paired
            datasets,
            enabling
            further comparison among different target lists or regions.
            <br>
            <img src="/sRNAanalyst_static/example/job/analysis/Fold_Change_0.png" class="mx-auto w-75 my-3">
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Scatter Plot:</b> Similar to Fold Change, but presented as scatter plots. This visualization can further
            annotate
            target RNAs with a p-value using statistical markers.
            <br>
            <img src="/sRNAanalyst_static/example/job/analysis/Scatter_0.png" class="mx-auto w-75 my-3">
          </li>
        </ul>
      </div>
    </div>
  </div>

  <div class="col-12" id="Database-Page">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">Database Page</h5>
        <p>
          The Database page allows users to access common literature data, primarily focusing on <i class="fw-bold">C.elegans</i> at present.
          Clicking on individual data fields enables users to download or view detailed information and external links,
          aiding in swiftly comparing their research findings with literature data.
        </p>
        <ul>
          <li class="my-1">
            <b>Dataset:</b> Compressed files containing NGS data such as WAGO-1 IP, HRDE-1 IP, CSR-1 IP, PRG-1 CLASH, ALG-1 iCLIP, etc.
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Reference:</b> Includes miRNAs from <i>C.elegans, Homo sapiens, Mus musculus, Drosophila melanogaster</i>, as well as some mRNA and ncRNA.
          </li>
        </ul>
        <ul>
          <li class="my-1">
            <b>Target List:</b> Based on the WormBase WS285 version for <i>C.elegans</i>, comprising all mRNAs, WAGO-1 targets, CSR-1 targets, Germline targets, Germline-enriched targets, and Somatic targets.
          </li>
        </ul>
      </div>
    </div>
  </div>

  <div class="col-12" id="Source-Code">
    <div class="card">
      <div class="card-body">
        <h5 class="card-title">Source Code</h5>
        <p>
          You can download the source code for the <b>Command Line tool</b> from here: <a href="https://github.com/RyanCCJ/sRNAanalyst" target="_blank">https://github.com/RyanCCJ/sRNAanalyst</a>
        </p>
        <p>
          If you prefer to use the <b>Docker version</b>, you can download it from here: <a href="https://github.com/RyanCCJ/sRNAanalyst-docker" target="_blank">https://github.com/RyanCCJ/sRNAanalyst-docker</a>
        </p>
      </div>
    </div>
  </div>

</div>